components/TicTacToe.jsx

```diff
@@ -7,22 +7,64 @@ export const TicTacToe = () => {
     border: "1px solid",
     boxSizing: "content-box",
   };
+  const cols = 3;
+  const rows = 3;
   const pieces = ["X", "O"];
   const [turn, setTurn] = useState(0);
   const [grid, setGrid] = useState(Array(9).fill(null));
+  const [win, setWin] = useState("in progress");
+
+  const notClicked = (i) => {
+    if (pieces.indexOf(grid[i]) < 0) {
+      return true;
+    }
+    return false;
+  };
```

Comment:
This is not a good function name, it's not clear what it does.

See: https://en.wikipedia.org/wiki/Self-documenting_code

Additionally, assuming you did want to use this function, the two return statements are inefficient / excessive.

1. Notice that the expression in the `if` clauses already returns a boolean.
2. If you're not doing anything special inside of the `if/else` blocks, return that expression directly.

```suggestion
  const notClicked = (i) => {
      return pieces.indexOf(grid[i]) < 0);
  };
```

Also:
- Why are you comparing it to `0`?
- It might seem to make more sense to rename this function `isEmpty`
- Why are you checking against `pieces`? That seems... roundabout. Use the most straightforward comparisons/references in code whenever possible. Just check `return grid[i] === null;`